### PR TITLE
Col plan advanced

### PR DESCRIPTION
**Pull request by thebioengineer**
Moving behavior of col_plan and span_structure to:

- col plan acts as a "what you said last is what you expect" situation, but with the order in which things are put changing the column order as we go
- span_structure updated to take multiple columns, the arg name representing the name of the column to apply to. 

**note** documentation has not been updated, as well as many of the tests. Opening this PR for discussion and iteration.

Look at test-col_plan.R for my tests, specifically `Test applying a col_plan - simple`, `Test applying a col_plan - renaming`, and `Test applying a col_plan - ordering on multiple columns` for examples of functioning tests. the `Unorthodox col_plans` has one functioning tests, the second one is failing at the moment but I wanted to chat about them.

### CODE REVIEW

**Comment by thebioengineer** [comment]
col_plan is functioning and tested as we all discussed. I am now working my way through the vignettes and examples across the package.

**Review by statasaurus — APPROVED** [review]
I like how you have broken things apart. I have only done a high level review but I am going to play more in the coming days

**Review by statasaurus — CHANGES_REQUESTED** [review]
still needs to work through all the bugs includings duplicate naming

File: R/col_plan.R [hunk 1]
```diff
@@ -21,21 +24,20 @@
...
 #' @param ... For a col_plan and span_structure,
 #'   <[`tidy-select`][dplyr_tidy_select]> arguments, unquoted expressions
-#'   separated by commas, and span_structures. Span_structures can nest
-#'   additional span_structures. To use a span_structure, there can only be one
-#'   defined "column" in the tfrmt.
-#' @param .drop Boolean. Should un-listed columns be dropped from the data. Defaults to FALSE.
+#'   separated by commas, and span_structures. span_structures must have the
+#'   arguments named to match the name the column in the input data has to identify the correct columns. See the examples
+#' @param .drop Boolean. Should un-listed columns be dropped from the data.
+#'   Defaults to FALSE.
```
[the remark this answers is not in the document]
Comment (thebioengineer) [reply R/col_plan.R:40]: I think based on our discussion that we are okay with it staying the default for now.

**Review by statasaurus — APPROVED** [review]
Looks greeeeaaaat 🐯 (think Tony the tiger)

**Pull request merged**